src/status_im/ui/components/invite/events.cljs

```diff
@@ -164,17 +165,23 @@
 
 (fx/defn starter-pack
   {:events [::starter-pack]}
-  [{:keys [db]}]
-  (let [contract (contracts/get-address db :status/acquisition)]
-    {::json-rpc/eth-call [{:contract   contract
-                           :method     "getDefaultPack()"
-                           :outputs    ["address" "uint256" "address[]" "uint256[]" "uint256[]"]
-                           :on-success #(re-frame/dispatch [::starter-pack-amount (vec %) (prn %)])}]}))
+  [{:keys [db]} contract]
+  {::json-rpc/eth-call [{:contract   contract
+                         :method     "getDefaultPack()"
+                         :outputs    ["address" "uint256" "address[]" "uint256[]" "uint256[]"]
+                         :on-success #(re-frame/dispatch [::starter-pack-amount (vec %) (prn %)])}]})
 
 (re-frame/reg-sub-raw
```

Comment:
could you please elaborate on this? why would you want to have raw subs and dispatch from them

Reply:
This is a nice approach to use just subscription for external data, this way we do not need to trigger events on some actions but they will be triggered when the data is requested by the subscription. There is a similar example in the docs https://github.com/day8/re-frame/blob/master/docs/Subscribing-To-External-Data.md#some-code

Reply:
i'm not sure i understand the reason why you can't dispatch event in the place where you set value in app-db ?  i mean the cool thing about re-frame its simple, and predictable, this one just looks like can be done in "regular" way but just adds one more case for devs to know an remember , not sure if it worth it

Reply:
for me it looks like unpredictable thing, i don't expect that event will be dispatched when i use subscription, instead i'll be looking to component-will-mount

Reply:
probably if we really want to use this approach at least we could give a names for such subscriptions, like `::get-starter-pack-and-dispatch`

Reply:
Well, that's described in re-frame documentation, so it is correct in terms of the re-frame flow. The problem with the dispatching event is that we show it in 3 places, and we need then to dispatch it every time before showing any of these screens. Also as a user of API, you should no care from where this data is resolved. Mostly you will look at this subscription as "give me the starter pack" and the subscription will give you the value. The underline implementation makes sense only if you want to change it. But as this event does involve data mutation it can be interpreted as a pure pulling of data.

Dispatch actually is not needed here at all, I can call the fx directly, the dispatch call is here because firstly it was implemented with dispatch during navigation to the invite screen, then refactored to raw when it became needed in other places. So I think I can remove the dispatch by replacing it with the fx call directly, and keep the name as it is.